idaes/core/util/scaling.py

```diff
@@ -462,6 +481,10 @@ def badly_scaled_var_generator(
     their current scale factors and values. For each potentially poorly scaled
     variable it returns the var and its current scaled value.
 
+    Note that while this method is a reasonable heuristic for nonnegative
```

Comment:
We're looking at the magnitude here, I don't think nonnegative maters.

Reply:
As you might be able to tell, I don't think this is a particularly reliable method for finding badly scaled variables. "Nonnegative" is trying to get at one of the failure conditions of this method: when you have a `Constraint` constraining a variable to equal zero. You have those with no-flux boundary conditions in the SOEC, and you might also get those if you want to have no pressure drop and constraint `DeltaP` to equal zero (for the moment, set aside whether or not fixing the variable to equal zero would be a better method). If you solve the problem and look at the scale of the variable at the solution, it's absolutely not going to give you a good scaling factor for the variable.

You also encounter this with enthalpy as a state variable. It has some zero point determined by the reference state, but can take on a wide range of negative and positive values. If you happen to have an enthalpy near zero, the scaling factor you need to give it isn't any different than the rest of the enthalpies.